libiocage/lib/Release.py

```diff
@@ -310,75 +369,129 @@ def update(self):
         dataset.snapshot(snapshot_name, recursive=True)
 
         jail = libiocage.lib.Jail.Jail({
-            "uuid"              : str(uuid.uuid4()),
-            "basejail"          : False,
-            "allow_mount_nullfs": "1",
-            "release"           : self.name,
-            "securelevel"       : "0"
-        },
+                "uuid"              : str(uuid.uuid4()),
+                "basejail"          : False,
+                "allow_mount_nullfs": "1",
+                "release"           : self.name,
+                "securelevel"       : "0"
+            },
+            new=True,
             logger=self.logger,
             zfs=self.zfs,
             host=self.host
         )
 
         jail.dataset_name = self.dataset_name
 
+        changed = False
+        try:
+            if self.host.distribution.name == "HardenedBSD":
+                changed = self._update_hbsd_jail(jail)
+            else:
+                changed = self._update_freebsd_jail(jail)
+        except:
+            self.logger.verbose(
+                "There was an error updating the Jail - reverting the changes"
+            )
+            jail.stop(force=True)
+            self.zfs.get_snapshot(snapshot_name).rollback(force=True)
+            raise
+
+        return changed
+
+
+    def _update_hbsd_jail(self, jail):
+
+        jail.start()
+
+        update_script_path = f"{self.release_updates_dir}/hbsd-update"
+        update_conf_path = f"{self.release_updates_dir}/hbsd-update.conf"
+
+        try:
+
+            # ToDo: as bad as print() - replace passthru with a nice progress bar
+            stdout = libiocage.lib.helpers.exec_iter([
+                update_script_path,
+                "-c",
+                update_conf_path,
+                "-j",
+                jail.identifier,
+                "-V"
+            ], logger=self.logger)
+
+            for stdout_line in stdout:
+                self.logger.verbose(stdout_line.strip("\n"), indent=1)
+
+            self.logger.debug(f"Update of release '{self.name}' finished")
+            changed = True
```

Comment:
Why define `changed`, as you never set it to anything but true, or raise. Just return `True`

Reply:
https://github.com/iocage/libiocage/issues/31 most certainly requires action after a release was updated. Left a ToDo note in the code that we need to parse the hbsd-update output and see if changes were applied to the release.